Allows non-welding tool welders to repair IPCs and prosthetics

benbot16 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This changes welding carbons to be handled by an element, and not the welding tool, allowing things with the welding tool tool behavior to repair mechanical limbs. It also moves cable repairs to said element as well.

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->
Accidentally delimbing IPCs with a plasma cutter sucks.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>


https://user-images.githubusercontent.com/9423435/191390413-21665072-5158-45d9-a004-f33366a95d4a.mp4


</details>

## Changelog
:cl:
fix: Repairing prosthetic limbs with non-welding tool welders works now
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

benbot16 committed:
Moves welding repair to carbon/attackby, allowing items with welding tool behavior to fix carbons

benbot16 committed:
Code change

Archanial reviewed (dismissed):
Putting it on the carbon is definitely a bad idea.
Ideally this could be made a component instead

benbot16 committed:
Moves mechanical repair to an element, instead of carbon code

I used an element instead of a component because we really only need one copy of this.

PowerfulBacon reviewed (dismissed):
Good PR

benbot16 committed:
Resolving conflicts
